bin/ci-builder

```diff
@@ -16,7 +16,7 @@
 
 set -euo pipefail
 
-NIGHTLY_RUST_DATE=2023-04-09
+NIGHTLY_RUST_DATE=2023-05-28
```

Comment:
can we pick a july nightly?

Reply:
Done!

Reply:
Unfortunately the latest nightly we can use is 2023-06-27 to avoid https://github.com/rust-lang/rust/issues/113152. This issue was fixed in `proc-macro2 v1.0.60` which we can bump in our deps, but as part of the CI image builder we install serveral `cargo-*` programs, and at least [cargo-about](https://github.com/EmbarkStudios/cargo-about/blob/main/Cargo.lock) has a version of `proc-macro2` in their `Cargo.lock` that is too old.

Reply:
rip, then lets just stick with the old nightly!